packages/components/pie-list/src/pie-list-item/index.ts

```diff
@@ -1,15 +1,91 @@
-import { LitElement } from 'lit';
-import { safeCustomElement } from '@justeattakeaway/pie-webc-core';
-import { type ListItemProps } from './defs';
+import { html, unsafeCSS, nothing } from 'lit';
+import { PieElement } from '@justeattakeaway/pie-webc-core/src/internals/PieElement';
+import { RtlMixin, safeCustomElement } from '@justeattakeaway/pie-webc-core';
+import { property } from 'lit/decorators.js';
+import type { PropertyValues } from 'lit';
+
+import styles from './list-item.scss?inline';
+import { defaultProps, type ListItemProps } from './defs';
+
+// Valid values available to consumers
+export * from './defs';
 
 const componentSelector = 'pie-list-item';
 
 /**
  * @tagname pie-list-item
+ *
+ * PIE List Item component for use within pie-list.
+ *
+ * @slot - Primary content of the list item. Falls back to `primaryText` when
+ * no light DOM content is provided.
+ * @slot leading - Optional leading content such as icons, avatars, or
+ * thumbnails.
+ * @slot secondary - Optional secondary text displayed beneath the primary
+ * content.
+ * @slot trailing - Optional trailing content such as meta text or controls
+ * (e.g., checkbox, radio, switch).
  */
 @safeCustomElement('pie-list-item')
-export class PieListItem extends LitElement implements ListItemProps {
-    // component logic
+export class PieListItem extends RtlMixin(PieElement) implements ListItemProps {
+    @property({ type: String })
+    public primaryText = defaultProps.primaryText;
+
+    @property({ type: Boolean })
+    public isSelected = defaultProps.isSelected;
+
+    @property({ type: Boolean, reflect: true })
+    public disabled = defaultProps.disabled;
+
+    connectedCallback () {
+        super.connectedCallback();
+
+        if (!this.hasAttribute('role')) {
+            this.setAttribute('role', 'listitem');
+        }
+
+        this._syncHostClasses();
+        this._syncAriaDisabled();
+    }
+
+    protected updated (changedProperties: PropertyValues) {
+        super.updated(changedProperties);
+
+        if (changedProperties.has('isSelected')) {
+            this._syncHostClasses();
+        }
+
+        if (changedProperties.has('disabled')) {
+            this._syncAriaDisabled();
+        }
+    }
+
+    private _syncAriaDisabled () {
+        if (this.disabled) {
+            this.setAttribute('aria-disabled', 'true');
+        } else {
+            this.removeAttribute('aria-disabled');
+        }
+    }
+
+    private _syncHostClasses () {
+        this.classList.toggle('is-selected', this.isSelected);
+    }
```

Comment:
is there a reason why we can't reflect the `isSelected` prop and use it for styles such as 

```
 @property({ type: Boolean, reflect: true})
    public isSelected = defaultProps.isSelected;

```
and style it as 

```
:host(isSelected) .c-listItem-primary span {
  color: var(--dt-color-content-default);
}
```